### ISSUE TITLE
Sub-vector inference for K-statistic (Kleibergen-Mavroeidis 2009)

### ISSUE THREAD

**Issue by ligon** [ligon/ManifoldGMM#41]
## Motivation

`GMMResult.k_statistic(theta_0)` and `GMMResult.k_statistic_bootstrap(theta_0)` (PRs #25/#26) give a *full-vector* Kleibergen (2005) test: H_0: theta = theta_0 with the full p = manifold-dim degrees of freedom.

For rank tests on PSDFixedRank — "is rank K-1 enough?" — this is awkward.  Full-vector K tests *a specific nuisance basin's* rank-restricted point.  Different basins give different K's.  Concrete observation on the K-Aggregators cereal data, K=3 het exp-link fit (this session):

| theta_0 construction                              | K_obs | df_K | p_K (chi^2)    |
|---------------------------------------------------|-------|------|----------------|
| truncate-K3 (K=3 fit's A, sigma_3 -> 1e-8)        | 207   | 28   | ~0             |
| pad-K2     (K=2 fit's A, append small ortho col)  | 104   | 28   | 9.6e-11        |

Both reject so the qualitative conclusion ("rank 2 insufficient") is robust here, but the *size* of K is basin-dependent.  In situations where the two heads disagree, full-vector K cannot adjudicate.

The principled object is the *sub-vector* K from Kleibergen & Mavroeidis (2009), *Econometric Theory* 25, "Weak instrument robust tests in GMM and the new Keynesian Phillips curve" — same K decomposition but profiling over the nuisance parameters:

  K_sv(beta_0) = min over eta of K(beta_0, eta)

with chi^2(p_beta) reference distribution where p_beta is the dim of the tested sub-vector only.

## Proposed API

```python
GMMResult.k_statistic_subvector(
    *,
    test_subset,           # callable theta -> theta_subset (tested), or indexing spec
    test_value,            # the H_0 value of the test_subset (e.g. zero column)
    profile_optimizer=None,   # optional pymanopt optimizer for the inner minimization
    ridge_condition=1e8,
) -> KStatisticResult
```

Semantics: minimize K(theta_0) over the nuisance manifold with `test_subset(theta_0) == test_value` enforced, return the minimum value with chi^2(p_test) p-value.

A bootstrap variant `k_statistic_subvector_bootstrap` parallels the full-vector bootstrap from #25, refitting each replicate's profile.

## Why this is hard (and why it's worth doing)

The hard part is the inner minimization: on a manifold-valued parameter, "fix one sub-block, profile over the rest" requires either:

(a) a product-manifold decomposition that exposes the test/nuisance split cleanly (works for product manifolds; awkward for PSDFixedRank where "fix sigma_K = 0" is a manifold boundary), or
(b) a numerical sweep with Lagrangian penalty enforcing the restriction.

The cleaner path may be (b): minimize K(theta) + rho * ||test_subset(theta) - test_value||^2 with rho growing.  rho -> infinity gives K_sv.

The pay-off is sizeable: any GMM rank test or restriction test under weak identification needs sub-vector inference.  This is the standard tool in weak-IV econometrics (Andrews & Mikusheva 2016 review for survey), and full-vector K is rarely the right object in practice.

## Concrete first use-case (driver)

K-Aggregators wants to test H_0: lambda_K(Gamma) = 0 on a PSDFixedRank(m, K) fit, profiling over the remaining (K-1) eigenvalues, eigenvectors, theta_block, and Pi.  Right now we approximate this with two basin-conditional full-vector K's (the table above); a sub-vector K would settle the question in one shot.

## References

- Kleibergen, F. (2005). "Testing Parameters in GMM Without Assuming that They Are Identified." *Econometrica* 73(4), 1103-1123.
- Kleibergen, F., & Mavroeidis, S. (2009). "Weak instrument robust tests in GMM and the new Keynesian Phillips curve." *Econometric Theory* 25, 70-103.
- Andrews, I., & Mikusheva, A. (2016). "Conditional inference with a functional nuisance parameter." *Econometrica* 84(4), 1571-1612.

## Empirical evidence this matters

K-Aggregators session 2026-05-23: \`reports/kleibergen_sanity_20260523.org\` documents the basin-sensitivity gap (K=207 vs K=104 on the same null) and the implementation-level confidence that the existing full-vector machinery is computing correctly.

**Comment by ligon** [comment]
## Design comment: Lagrangian-via-penalty-hook approach, MR split, talk-window plan

Endorsing the Lagrangian-penalty path proposed in the body (option (b)).  Endorsement is strong because the existing #19 MR1 penalty hook is **structurally identical** to what the Lagrangian needs -- the inner minimisation reduces to running an ordinary `GMM`-like fit with a parameter-space penalty enforcing the test constraint.  Composable rather than bespoke.

### Why Lagrangian over product-manifold decomposition

The K-Aggregators driver (rank test on PSDFixedRank(m, K)) is where the decomposition approach (a) breaks: "fix sigma_K = 0" puts the iterate on the *boundary* of PSDFixedRank(m, K), where rank-(K-1) matrices actually live on PSDFixedRank(m, K-1) -- a different manifold.  Product-manifold decomposition has nothing to say about that.  Lagrangian-penalty bypasses the boundary problem entirely: minimise `K(theta) + rho * ||test_subset(theta) - test_value||^2` on the full original manifold; for large rho the iterate approaches the constraint surface (rank-K-1 boundary) smoothly without ever leaving the chart.

### MR split for the implementation

**MR1 (Euclidean nuisance, no gauge interaction).**

- New method `GMMResult.k_statistic_subvector(*, test_subset, test_value, profile_optimizer=None, rho_schedule=..., ridge_condition=1e8) -> KStatisticResult`.
- `test_subset(theta) -> ndarray`: callable extracting the tested coords (must be JAX-traceable).
- Inner minimisation: builds a fresh `GMM`-shaped problem whose criterion is `K(theta)` (using the same `k_statistic` formula machinery but as a function of theta, not just at a single eval_point) plus the Lagrangian penalty, and runs the same `LoggingTrustRegions` machinery.
- ``rho_schedule`` defaults to geometric increase (e.g.\\ start at ``rho_0 = max_eig(D'WD)``, double each outer iteration of the penalty method until ``K_obs`` stabilises to within tolerance or a hard cap is hit).  Caller can override.
- Returned ``KStatisticResult`` carries ``K``, ``df_K = len(test_value)`` (NOT the full p), ``p_K`` from ``chi^2(df_K)``.
- Tests: Euclidean linear-IV fixture with a known closed-form sub-vector K (where the test slice's null distribution is analytically chi^2).  Bootstrap-vs-asymptotic agreement on a well-conditioned case.

**MR2 (general manifolds + gauge composition).**

- Extend MR1 to handle ``test_subset`` returning a value on a sub-manifold (e.g., the smallest eigenvalue of a PSD block).
- Compose with #32's ``_resolve_gauge_dim``: the *nuisance* sub-manifold can itself have gauge (e.g., the remaining rank-(K-1) Gamma block has its own ``O(K-1)`` rotation orbit), and the inner profile's ``D'WD`` needs ``exclude_gauge`` semantics applied to the *nuisance* directions.  Probably reuse ``_gauge_dim_of_manifold`` on the nuisance sub-manifold detected via the test/nuisance split.
- K-Aggregators rank-test driver lands here.

**MR3 (bootstrap variant).**

- ``k_statistic_subvector_bootstrap`` parallel to PR #27's ``k_statistic_bootstrap``: cluster-wild Rademacher signs on centred moments, refit the *profile* per replicate, accumulate K* distribution.
- Cost is the killer here: each replicate runs a full Lagrangian penalty schedule.  200 replicates x ~50-200 inner iterations x ridge_inverse + JAX traces = wall-clock hours per call.  Probably need #38's ``run_parallel`` + #37's JIT cost to be tolerable.  Warm-start the profile from the previous replicate's solution where possible.

### Open design questions (not blockers, but worth thinking about before MR1 lands)

1. **``K(theta)`` as a JAX-traceable function.**  The current ``k_statistic`` implementation calls ``ridge_inverse`` (NumPy-only) twice.  For the Lagrangian inner minimisation to use JAX autodiff cleanly, ``K`` needs to be re-expressible in pure JAX (e.g., via ``jnp.linalg.solve`` with the same ridge logic).  Alternative: FD gradients.  Likely a small refactor; check on MR1 entry.
2. **rho schedule convergence detection.**  "K_obs stabilised" is fuzzy.  Concrete predicate: ``|K(rho_n) - K(rho_{n-1})| / max(1, K(rho_n)) < tol`` with ``tol`` defaulted to 1e-3, hard cap on outer rho-iterations (10-20).  Document on the docstring.
3. **Composition with the penalty hook from #19.**  If the *original* fit was penalised, the sub-vector K computation still inherits the penalty-independence property from PR #26 (K at theta_0 doesn't depend on the optimiser-side penalty).  Both penalties can compose at the Lagrangian level: ``K(theta) + rho * constraint_pen(theta) + original_pen(theta)``.  Tested cleanly in MR1.
4. **Initial point.**  Inner profile starts from ``GMMResult.theta_point`` (the original fit's theta_hat).  This is the obvious warm-start; on rank-test cases where the test forces theta away from theta_hat substantially, the inner solver may need many iterations to find the constrained optimum.  Document; consider exposing ``initial_point`` kwarg.

### Recommendation for the 2026-05-26 talk

Even MR1 alone is 2-3 days of careful implementation + math review.  Not ship-by-talk territory.  Concrete recommendation for K-Aggregators talk materials:

- **Report both basin-conditional full-vector K-statistics** (the table in this issue's body) with a footnote noting their basin-dependence, citing Kleibergen-Mavroeidis (2009) for the principled sub-vector object, and indicating that sub-vector inference is in the implementation pipeline (this issue).
- **The qualitative conclusion ("rank 2 insufficient") is robust** on the current data because both basin-conditional Ks reject -- mention this explicitly in the footnote.  In a closer call the two would disagree and neither would be principled, which is exactly why the upstream feature is needed.

This is defensible.  It cites the right reference, acknowledges the limitation, points at the planned fix.  Talk-quality.

Post-talk: MR1 lands when there's time for proper math review.  No rush.

### Cross-references

- **#19 MR1** (PR #22, merged): the penalty hook this PR's Lagrangian formulation reuses verbatim.  ``constraint_pen(theta) = lambda * ||test_subset(theta) - test_value||^2`` is exactly the kind of penalty the hook was designed for.
- **#32 / PR #33** (merged): ``_gauge_dim_of_manifold`` + threshold detection are reused in MR2 for the nuisance-side gauge.
- **#25 / PR #27** (merged): full-vector ``k_statistic_bootstrap`` is the structural sibling of MR3's bootstrap variant.
- **#37, #38** (merged): JIT cost + parallel bootstrap make the otherwise-prohibitive MR3 bootstrap tractable.
- **Kleibergen, F. & Mavroeidis, S. (2009)**, *Econometric Theory* 25, 70-103 (the reference).
- **Andrews, I. & Mikusheva, A. (2016)**, *Econometrica* 84(4), 1571-1612 (broader weak-IV survey).
